qiskit_ibm_provider/job/ibm_circuit_job.py

```diff
@@ -114,6 +120,8 @@ def __init__(
         job_id: str,
         creation_date: str,
         status: str,
+        runtime_client: RuntimeClient = None,  # TODO: make mandatory after completely switching
+        result_decoder: Any = None,
```

Comment:
We don't really need `result_decoder`, as it should always return a `Result`.

Reply:
Done